### DISCUSSION TITLE
Should the `virtualdefund.Objective` embed `V *channel.Channel`?

### DISCUSSION THREAD

**Discussion by geoknee** [statechannels/go-nitro#617] [General]
Against:
* virtual channels are payment channels. Payment channels may be considered special, because we may hand off responsibility for payments to the consuming application. So then the wallet doesn't store the latest payment, and needs to get this from the consuming application. 

For:
* all of the other objectives embed at least one channel
* it fits the "old" payment model well, where the wallet stores the latest payment (= state update). The objective can read this from the embedded channel.
* if a challenge is required or detected, the wallet cannot simply query the channel for the information it needs. It has to figure out whether a virtual defund objective is running (this is a special case), and if so query that, else query the channel.

**Comment by lalexgap** [comment]
@andrewgordstewart outlined some benefits he sees [here](https://github.com/statechannels/go-nitro/pull/503#discussion_r846321196):

> Doing it this way has some benefits:
> 
> 1. unit tests for the VDO are easier to write (you don't need to construct a valid `V`, and you have one fewer dependency to worry about)
> 2. there's one fewer relation, which means: one less entry in `Related()`, cloned, json marshalled, diffed, etc. these things add up over time!
> 3. (theoretical) there would be queries required against a store during the defunding process, if we were actually querying against a store
> 4. code is more explicit: rather than asking "is the latest supported state in `V` final, and does it have the correct outcome and turn number?" the VDO would ask "do I have three (valid) signatures on the final state?"
>

**Comment by lalexgap** [comment]
>if a challenge is required or detected, the wallet cannot simply query the channel for the information it needs. It has to figure out whether a virtual defund objective is running (this is a special case), and if so query that, else query the channel.

If a challenge occurs would we need to query the objectives anyways to pause/cancel the objective? I think we would want to transfer ownership of the channel from any existing objective to some kind of challenge objective.

My vote is for just updating virtual defund to use a `Channel` like the rest of our objectives. It makes the code easier to understand and modify if we treat channels consistently in objectives.
